### ISSUE TITLE
`pyupgrade` rule: upgrading `str, Enum` to `StrEnum`

### ISSUE THREAD

**Issue by jamesbraza** [astral-sh/ruff#9569]
Since Python 3.11 there are now `StrEnum` and `IntEnum`:

```python
from enum import Enum, IntEnum, StrEnum

class Foo(str, Enum):  ...  # Python<3.11
class Foo(StrEnum):  ...  # Python>=3.11

class Bar(int, Enum): ...  # Python<3.11
class Bar(IntEnum): ...  # Python>=3.11
```

`pyupgrade` can be used to suggest this change.

However, this request was rejected in `pyupgrade` itself here: https://github.com/asottile/pyupgrade/issues/779. Apparently they're not identical in behavior. Luckily, `ruff` has a concept of safe vs unsafe, so I think `ruff` can actually implement this rule.

The request here is for `ruff` to suggest this upgrade as an unsafe fix, ideally with autofix support.

**Comment by zanieb** [comment]
Sounds good to me!

**Comment by bswck** [comment]
Side note here: `IntEnum` was already available since Python 3.4. Only `StrEnum` came in 3.11.

**Comment by arferreira** [comment]
Hey @zanieb, is this issue active? Can I try to upgrade this?

**Comment by charliermarsh** [comment]
Go for it @arferreira!

**Comment by dhruvmanila** [comment]
Relevant discussion: https://github.com/astral-sh/ruff/discussions/3867
